### ISSUE TITLE
Effect doesn't respect rippleContainerBorderRadius on Android 

### ISSUE THREAD

**Issue by perrosnk** [n4kz/react-native-material-ripple#2]
I created a circular container, inside it I placed `<Ripple/>`, and I set the `rippleContainerBorderRadius` equal to the `borderRadius` of the container. Although on iOS, the border is respected, on Android it's not. Has anyone experienced the same problem?

**Comment by n4kz** [comment]
As described [here](https://facebook.github.io/react-native/releases/0.26/docs/known-issues.html#the-overflow-style-property-defaults-to-hidden-and-cannot-be-changed-on-android) `overflow:hidden` is not working with `borderRadius` on Android. Corresponding issue facebook/react-native#3198 is not yet resolved. However, if you need round controls, you can set `rippleCentered` and `rippleSize`.

```javascript

const radius = 20;

class CircleButton extends Component {
  render() {
    return (
      let { children, ...props } = this.props;
      <Ripple
        style={{ width: radius * 2, height: radius * 2, borderRadius: radius }}
        rippleCentered={true}
        rippleSize={radius * 2}
        {...props}
      >
        {children}
      </Ripple>
    );
  }
}
```

**Comment by perrosnk** [comment]
`rippleCenteredz is not working on my App, neither on iOS or Android. Is it working in yours?

**Comment by n4kz** [comment]
Yes, works as expected on both platforms (ripple always starts from component center).

Take look at [example](https://github.com/n4kz/react-native-material-ripple/blob/master/example/app.js#L101)

**Comment by perrosnk** [comment]
My code:

```javascript
          <Ripple
            rippleContainerBorderRadius={90}
            rippleColor={'#FBB829'}
            rippleCentered={true}
            rippleSize={180}
            style={styles.mainButton}>
            <Text>
              Hi
            </Text>
          </Ripple>

const styles = StyleSheet.create({
  mainButton: {
    width: 180,
    height: 180,
    borderRadius: 90,
    alignItems: 'center',
  },
})

```

Ripple is not centered.
Which RN version are you using?

**Comment by n4kz** [comment]
Tested on RN 0.41 and 0.42, `rippleCentered` was added in 0.4.0, please check your react-native-material-ripple version.

**Comment by perrosnk** [comment]
I am using RN 0.38, and react-native-material-ripple 0.2.0. So that must be the problem. Does 0.4.0 support RN 0.38?

**Comment by n4kz** [comment]
Should be safe to update to recent version (0.4.2).

**Comment by perrosnk** [comment]
I would suggest you don't close this issue till RN solves its issue, and your package behaves on Android the same way it does on iOS.

**Comment by carlbenson** [comment]
I still see this problem on Android. Version 0.7.2 on RN 0.44.2

**Comment by n4kz** [comment]
Yep, still not fixed in RN.

**Comment by sergeylaptev** [comment]
It seems to be fixed with RN 0.50

**Comment by n4kz** [comment]
Thanks for great news! I will make a new release on demand when RN 0.50 goes stable.
